gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    private Extension buildProblemSignificance(CD linksetCode) {
+        Optional<String> code = Optional.ofNullable(linksetCode.getQualifier().get(0).getName().getCode());
+        Extension extension = new Extension();
+        extension.setUrl(PROBLEM_SIGNIFICANCE_URL);
+
+        if (code.isPresent() && code.get().equals(MAJOR_CODE)) {
+            extension.setValue(new CodeType("Major"));
+        } else {
+            extension.setValue(new CodeType("Minor"));
+        }
+
+        return extension;
+    }
+
+    private Optional<Condition.ConditionClinicalStatus> buildClinicalStatus(CD linksetCode) {
+        if (linksetCode.getCode().equals(CLINICAL_STATUS_ACTIVE_CODE)) {
+            return Optional.of(Condition.ConditionClinicalStatus.ACTIVE);
+        } else if (linksetCode.getCode().equals(CLINICAL_STATUS_INACTIVE_CODE)) {
+            return Optional.of(Condition.ConditionClinicalStatus.INACTIVE);
+        }
+        return Optional.empty();
+    }
+
+    private CodeableConcept generateCategory() {
+        Coding coding = new Coding();
+        coding.setSystem("https://fhir.hl7.org.uk/STU3/CodeSystem/CareConnect-ConditionCategory-1");
+        coding.setCode("problem-list-item");
+        coding.setDisplay("Problem List Item");
+
+        return new CodeableConcept()
+            .addCoding(coding);
+    }
+
+    private List<Extension> buildRelatedClinicalContent(List<Resource> relatedClinicalResources) {
+        return relatedClinicalResources.stream()
+            .map(ConditionMapper::buildResourceReference)
+            .map(reference -> buildReferenceExtension(RELATED_CLINICAL_CONTENT_URL, reference))
+            .collect(Collectors.toList());
+    }
+
+    private List<Annotation> buildNotes(Optional<RCMRMT030101UK04ObservationStatement> observationStatement, RCMRMT030101UK04LinkSet linkSet) {
+        List<Annotation> annotationList = new ArrayList<>();
+
+        observationStatement.ifPresent(observationStatement1 -> observationStatement1.getPertinentInformation()
+            .stream()
+            .map(RCMRMT030101UK04PertinentInformation02::getPertinentAnnotation)
+            .filter(Objects::nonNull)
+            .map(RCMRMT030101UK04Annotation::getText)
+            .filter(StringUtils::isNotBlank)
+            .map(StringType::new)
+            .map(Annotation::new)
+            .forEach(annotationList::add));
+
+        if (linkSet.getCode().getOriginalText() != null) {
+            StringType text = new StringType(linkSet.getCode().getOriginalText());
+            annotationList.add(new Annotation(text));
+        }
+
+        return annotationList;
+    }
+
+    private static Reference buildResourceReference(Resource resource) {
+        IdType idType = new IdType(resource.getResourceType().name(), resource.getId());
+        return new Reference(idType);
+    }
+
+    private Extension buildReferenceExtension(String url, Reference reference) {
```

Comment:
Would it make sense to create a util class for this and `buildResourceReference` methods? It seems like those operations will happen often inside the mappers. Do you think extracting these to a separate class could be useful?